src/main/java/betterquesting/api2/client/gui/panels/lists/CanvasQuestSearch.java

```diff
@@ -62,16 +61,31 @@ private QuestSearchEntry createQuestSearchEntry(DBEntry<IQuestLineEntry> iQuestL
 
     @Override
     protected void queryMatches(QuestSearchEntry entry, String query, ArrayDeque<QuestSearchEntry> results) {
+        // claimable quests
         if ("@complete".startsWith(query) && 1 < query.length()) {
             if (entry.getQuest().getValue().isComplete(questingUUID) && entry.getQuest().getValue().canClaim(player)) {
                 results.add(entry);
             }
-        } else if (String.valueOf(entry.getQuest().getID()).contains(query)) {
-            results.add(entry);
-        } else if (entry.getQuest().getValue().getProperty(NativeProps.NAME).toLowerCase().contains(query)) {
-            results.add(entry);
-        } else if (QuestTranslation.translate(entry.getQuest().getValue().getProperty(NativeProps.NAME)).toLowerCase().contains(query)) {
+        } else if (
+            // quest id
+            String.valueOf(entry.getQuest().getID()).contains(query)
+            // quest title
+            || entry.getQuest().getValue().getProperty(NativeProps.NAME).toLowerCase().contains(query)
+            || QuestTranslation.translate(entry.getQuest().getValue().getProperty(NativeProps.NAME)).toLowerCase().contains(query)
+            // quest desc
+            || entry.getQuest().getValue().getProperty(NativeProps.DESC).toLowerCase().contains(query)
+            || QuestTranslation.translate(entry.getQuest().getValue().getProperty(NativeProps.DESC)).toLowerCase().contains(query)
+        ) {
             results.add(entry);
+        } else {
+            for (DBEntry<ITask> task : entry.getQuest().getValue().getTasks().getEntries()) {
+                if (task.getValue().getTextsForSearch() == null) continue;
+                for (String text : task.getValue().getTextsForSearch()) {
+                    if (text.toLowerCase().contains(query)) {
```

Comment:
To avoid creating in-memory copies of all texts here, you can use `org.apache.commons.lang3.StringUtils.containsIgnoreCase(text, query)`

Reply:
oh good